core/src/avm2/globals.rs

```diff
@@ -83,7 +83,8 @@ pub struct SystemClasses<'gc> {
     pub sprite: ClassObject<'gc>,
     pub simplebutton: ClassObject<'gc>,
     pub regexp: ClassObject<'gc>,
-    pub vector: ClassObject<'gc>,
+    pub generic_vector: ClassObject<'gc>,
+    pub object_vector: ClassObject<'gc>, // Vector.<*>
```

Comment:
Just checking- is this comment on the correct line? Isn't `Vector.<*>` the generic vector?

Reply:
Actually no.
`generic_vector` refers to `Vector` itself. This is the only type with the `GENERIC` flag.
`object_vector` refers to `Vector<*>` (which is different but almost-identical to `Vector<Object>`).

Feel free to offer alternative naming :D

Reply:
`untyped_vector` (matches `*` definition on https://web.archive.org/web/20171020231504/https://help.adobe.com/en_US/LiveCycleMosaic/9.0/actionscript/specialTypes.html) instead of `object_vector` maybe.

And just to avoid confusion, maybe `general_vector` instead of `generic_vector`, since in other langauges the star is called a generic.

Reply:
Maybe `typed_vector` and `untyped_vector`? (If I understand the difference correctly!)

Reply:
Scratch my suggestion I think, in the case of vectors the definition of the `*` doesn't match that page.

Reply:
Left the names as-is, expanded the comments.